File: app/src/main/java/com/example/languagegym/data/Word.java
package com.example.languagegym.data;

import java.util.List;

public class Word {
    private String word;
    private String translation;
    private String partOfSpeech;
    private String gender;
    private List<String> declension;
    private List<String> synonyms;

    public Word(String word, String translation, String partOfSpeech, String gender, List<String> declension, List<String> synonyms) {
        this.word = word;
        this.translation = translation;
        this.partOfSpeech = partOfSpeech;
        this.gender = gender;
        this.declension = declension;
        this.synonyms = synonyms;
    }

    public String getWord() {
        return word;
    }

    public void setWord(String word) {
        this.word = word;
    }

    public String getTranslation() {
        return translation;
    }

    public void setTranslation(String translation) {
        this.translation = translation;
    }

    public String getPartOfSpeech() {
        return partOfSpeech;
    }

    public void setPartOfSpeech(String partOfSpeech) {
        this.partOfSpeech = partOfSpeech;
    }

    public String getGender() {
        return gender;
    }

    public void setGender(String gender) {
        this.gender = gender;
    }

    public List<String> getDeclension() {
        return declension;
    }

    public void setDeclension(List<String> declension) {
        this.declension = declension;
    }

    public List<String> getSynonyms() {
        return synonyms;
    }

    public void setSynonyms(List<String> synonyms) {
        this.synonyms = synonyms;
    }
}

File: app/src/main/java/com/example/languagegym/data/Word.kt
package com.example.languagegym.data

import android.os.Parcel
import android.os.Parcelable
import kotlinx.android.parcel.Parcelize
import kotlinx.parcelize.Parceler


@Parcelize
data class Word(
    var word: String,
    var translation: String,
    var partOfSpeech: String,
    var gender: String,
    var declension: List<String>,
    var synonyms: List<String>,
    val imageUrl: String?,
    val learningProgress: Int
): Parcelable {
    companion object : Parceler<Word> {

        override fun Word.write(dest: Parcel, flags: Int) {
            dest.writeString(word)
            dest.writeString(translation)
            dest.writeString(partOfSpeech)
            dest.writeString(gender)
            dest.writeStringList(declension)
            dest.writeStringList(synonyms)
            dest.writeString(imageUrl)
            dest.writeInt(learningProgress)
        }

        override fun create(parcel: Parcel): Word {
            return Word(parcel)
        }
    }

    constructor(parcel: Parcel) : this(
        parcel.readString()!!,
        parcel.readString()!!,
        parcel.readString()!!,
        parcel.readString()!!,
        parcel.createStringArrayList()!!,
        parcel.createStringArrayList()!!,
        parcel.readString(),
        parcel.readInt()
    )

}